### PR TITLE
Add benefits pages to burials form

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3148

Adds benefits selection pages:

screen shot 2017-06-15 at 10 59 15 am
screen shot 2017-06-15 at 10 59 27 am
screen shot 2017-06-15 at 10 59 21 am
screen shot 2017-06-15 at 9 26 24 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -192,6 +192,6 @@
   "dependencies": {
     "camelcase-keys-recursive": "^0.8.2",
     "reselect": "^2.5.4",
-    "vets-json-schema": "https://github.com/department-of-veterans-affairs/vets-json-schema.git#175be45a2005e1d2ade7c43419d3dd0ca82d6c07"
+    "vets-json-schema": "https://github.com/department-of-veterans-affairs/vets-json-schema.git#689259b3a3134ed044f7b9edadd5f77201287a4f"
```
Comment (jbalboni) [inline package.json]: This is not in master yet.

Comment (jbalboni) [inline package.json]: This is not in master yet.

**Comment by jbalboni** [comment]
I still need to check on the explanatory text that's in the outline, I'm not sure how that is supposed to work and may be a separate ticket.

File: src/js/burials/config/form.js [hunk 2]
```diff
@@ -146,6 +161,152 @@ const formConfig = {
...
+            },
+            burialCost: {
+              'ui:title': 'Actual burial cost',
+              'ui:options': {
+                expandUnder: 'burialAllowanceRequested',
+                expandUnderCondition: 'vaMC'
+              }
+            },
+            previouslyReceiveAllowance: {
+              'ui:title': 'Did you previously recieve a VA burial allowance?',
```
Comment (cehsu) [inline src/js/burials/config/form.js]: receive not receive

File: src/js/burials/config/form.js [hunk 3]
```diff
@@ -146,6 +161,152 @@ const formConfig = {
...
+              }
+            }
+          },
+          schema: {
+            type: 'object',
+            required: ['burialAllowanceRequested'],
+            properties: {
+              burialAllowanceRequested,
+              burialCost,
+              previouslyReceiveAllowance,
```
Comment (cehsu) [inline src/js/burials/config/form.js]: If changed in backend schema, this should be updated as well.

**Comment by cehsu** [comment]
@jbalboni Good question about the explanatory text, is it intended to be a tooltip?

**Comment by cehsu** [comment]
Could you please swap out the burial allowance screenshot for one in which VA MC Death is selected to reveal the actual burial cost field?

**Comment by jbalboni** [comment]
@cehsu Yep, I asked and it's a link that opens a modal (the rx app does this). We would need to add support for that to our FieldTemplate component. I'm going to put that in a separate ticket.

**Comment by jbalboni** [comment]
@cehsu Ticket is here: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3332. It's ready to go if you want to tackle it after the burials page you're working on.

File: test/burials/config/plotAllowance.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      <DefinitionTester
+          definitions={formConfig.defaultDefinitions}
+          schema={schema}
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = getFormDOM(form);
+
+    formDOM.fillData('#root_placeOfRemains', 'Cemetary');
```
Comment (cehsu) [inline test/burials/config/plotAllowance.unit.spec.jsx]: I asked a question in the issue about whether any of the options here are mutually exclusive.

File: test/common/schemaform/validation.unit.spec.js [hunk 5]
```diff
@@ -339,4 +340,39 @@ describe('Schemaform validations', () => {
...
+    });
+
+    it('should add error if empty object', () => {
+      const errors = { addError: sinon.spy() };
+      validateBooleanGroup(errors, {});
+
+      expect(errors.addError.called).to.be.true;
+    });
+
+    it('should not add error if no props are true', () => {
```
Comment (cehsu) [inline test/common/schemaform/validation.unit.spec.js]: should this be is prop is true?

**Review by cehsu — APPROVED** [review]
LGTM. Thanks for already answering my questions. Had minor questions/comments.

**Review by cehsu — CHANGES_REQUESTED** [review]
It looks like some updates are required due to this new information from @stacyhlinn (copying from issue):
@cehsu

Let's update to:

Was the Veteran buried in a national cemetery, or one owned by the federal government?* [Y/N]

{If previous answer is N} Was the Veteran buried in a state Veteran's cemetery?* [Y/N]

cc'ing @J-Quag for transparency.

File: test/burials/config/burialAllowance.unit.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      <DefinitionTester
+          definitions={formConfig.defaultDefinitions}
+          schema={schema}
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = getFormDOM(form);
+    submitForm(form);
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error')).length).to.equal(1);
```
Comment (cvalarida) [inline test/burials/config/burialAllowance.unit.spec.jsx]: In the next test, you get the length of a `.querySelectorAll()` without doing `Array.from()`. Is it necessary?

File: test/burials/config/burialAllowance.unit.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = getFormDOM(form);
+
+    formDOM.fillData('#root_burialAllowanceRequested_2', 'vaMC');
+
+    submitForm(form);
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error')).length).to.equal(0);
```
Comment (cvalarida) [inline test/burials/config/burialAllowance.unit.spec.jsx]: Here too.

File: test/burials/config/plotAllowance.unit.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      <DefinitionTester
+          definitions={formConfig.defaultDefinitions}
+          schema={schema}
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = getFormDOM(form);
+    submitForm(form);
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error')).length).to.equal(4);
```
Comment (cvalarida) [inline test/burials/config/plotAllowance.unit.spec.jsx]: Flagging this one too for visibility...

File: test/burials/config/plotAllowance.unit.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    const formDOM = getFormDOM(form);
+
+    formDOM.fillData('#root_placeOfRemains', 'Cemetary');
+    formDOM.fillData('#root_federalCemetaryYes', 'Y');
+    formDOM.fillData('#root_stateCemetaryNo', 'N');
+    formDOM.fillData('#root_govtContributionsYes', 'Y');
+    formDOM.fillData('#root_amountGovtContribution', '50');
+
+    formDOM.submitForm(form);
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error')).length).to.equal(0);
```
Comment (cvalarida) [inline test/burials/config/plotAllowance.unit.spec.jsx]: Please forgive the [spam](https://www.youtube.com/watch?v=anwy2MPT5RE).

**Review by cvalarida — APPROVED** [review]

File: test/burials/config/burialAllowance.unit.spec.jsx (hunk 6 above)
Comment (cehsu) [inline test/burials/config/burialAllowance.unit.spec.jsx]: The NodeList should have a length property already? [source](https://www.w3schools.com/jsref/met_document_queryselectorall.asp)

File: src/js/burials/config/form.js [hunk 10]
```diff
@@ -146,6 +161,152 @@ const formConfig = {
...
+            'ui:title': 'Plot or Interment Allowance',
+            placeOfRemains: {
+              'ui:title': 'Place of Burial or Location of Deceased Veteran’s Remains'
+            },
+            federalCemetary: {
+              'ui:title': 'Was the Veteran buried in a national cemetery, or one owned by the federal government?',
+              'ui:widget': 'yesNo'
+            },
+            stateCemetary: {
+              'ui:title': 'Was the Veteran buried in a state Veteran’s cemetery?',
```
Comment (cehsu) [inline src/js/burials/config/form.js]: `Veterans` not `Veteran's`, courtesy of @peggygannon

File: test/burials/config/burialAllowance.unit.spec.jsx (hunk 6 above)
Comment (jbalboni) [inline test/burials/config/burialAllowance.unit.spec.jsx]: I think we did that to use it with `filter`, but yeah, not necessary here.

**Comment by jbalboni** [comment]
@cehsu Should be all fixed now.

I did realize I spelled cemetery wrong for the property names, so I'll make an issue to go back and fix that.

**Review by cehsu — APPROVED** [review]

**Pull request merged**